cms/templatetags/cms_admin.py

```diff
@@ -78,18 +128,72 @@ def get_page_display_name(cms_page):
     return cms_page.get_slug(language)
 
 
-@register.simple_tag(takes_context=True)
-def tree_publish_row(context, page, language):
-    cls = "cms-pagetree-node-state cms-pagetree-node-state-empty empty"
-    text = _("no content")
+class TreePublishRow(Tag):
```

Comment:
In the future the navigation part of the Page Admin should probably be in the menu app.